src/soundsourceproxy.cpp

```diff
@@ -248,51 +193,53 @@ void SoundSourceProxy::loadPlugins() {
         }
     }
 
-    s_soundSourceProviders.finishRegistration();
-
     const QStringList supportedFileExtensions(
-            s_soundSourceProviders.getSupportedFileExtensions());
-    foreach (const QString &supportedFileExtension, supportedFileExtensions) {
-        const Mixxx::SoundSourceProviderPointer pProvider(
-                s_soundSourceProviders.getProviderForFileExtension(supportedFileExtension));
-        const Mixxx::SoundSourcePluginLibraryPointer pPluginLibrary(
-                s_soundSourceProviders.getPluginLibraryForFileExtension(supportedFileExtension));
-        if (pPluginLibrary) {
-            qDebug() << "SoundSourceProvider for" << supportedFileExtension
-                    << "is" << pProvider->getName()
-                    << "@" << pPluginLibrary->getFilePath();
-        } else {
-            qDebug() << "SoundSourceProvider for" << supportedFileExtension
-                    << "is" << pProvider->getName();
+            s_soundSourceProviders.getRegisteredFileExtensions());
+    for (const auto &supportedFileExtension: supportedFileExtensions) {
+        qDebug() << "SoundSource providers for file extension" << supportedFileExtension;
+        const auto registrationsForFileExtension(
+                s_soundSourceProviders.getRegistrationsForFileExtension(
+                        supportedFileExtension));
```

Comment:
Sorry, I am struggling with this notation. The return type of getRegistrationsForFileExtension is out of sight and
GitHub has no tooltip for it. You may argue that the real type does not add any value, because it is used below as it is a container. But below you use auto as well (which would be nice if the reader knows the content).

getRegistrationsForFileExtension returns SoundSourceProviderRegistrationList which is a typedef that saves 3 characters and also hides information. I would prefer to use here QList<SoundSourceProviderRegistration>  directly. 

```
        const QList<SoundSourceProviderRegistration> registrationsForFileExtension(
                s_soundSourceProviders.getRegistrationsForFileExtension(
                        supportedFileExtension));
        for (const auto& registration: registrationsForFileExtension) {
 ```